### PR TITLE
DHCP Allows selecting the DHCP server's network interface

### PR DESCRIPTION
**Pull request by Governa**
Simple change allowing the selection of network interface on server start.

### CODE REVIEW

**Comment by kanflo** [comment]
Looks good to me.

File: extras/dhcpserver/include/dhcpserver.h [hunk 1]
```diff
@@ -26,7 +26,19 @@ extern "C" {
...
+
+/* Start DHCP server
+
+   See @dhcpserver_start_on_if.
+
+   Selects the first netif in netif_list as the server_if
+ */
+static inline void dhcpserver_start(const ip_addr_t *first_client_addr, uint8_t max_leases)
+{
+	dhcpserver_start_on_if(first_client_addr, max_leases, NULL);
```
Comment (ourairquality) [inline extras/dhcpserver/include/dhcpserver.h]: I suggest just changing `dhcpserver_start` to take the `netif` and patching all the callers to pass in the appropriate netif which they can get using `sdk_system_get_netif`. Perhaps even place the `netif` as the first argument. That seems better than reaching into the lwip `netif_list` and taking the first element from that list and separated the dhcp server a little more from the internals of lwip.

Comment (Governa) [inline extras/dhcpserver/include/dhcpserver.h]: Ok, I can do that. But is it ok to break the API like this?

**Comment by UncleRus** [comment]
Closing/reopening to trigger CI build

**Comment by ourairquality** [comment]
Commit 5ee6c548 needs to be reverted.

**Pull request merged**